src/repl-frontend/scala/tools/nsc/interpreter/shell/ILoop.scala

```diff
@@ -529,26 +530,21 @@ class ILoop(config: ShellConfig, inOverride: BufferedReader = null,
 
   def lineCommand(what: String): Result = editCommand(what, None)
 
-  def newCompleter(): ReplCompletion =
-    new ReplCompletion(intp) {
-      override def shellCompletion(buffer: String, cursor: Int): Option[CompletionResult] =
-        if (buffer.startsWith(":")) Some(colonCompletion(buffer, cursor).complete(buffer, cursor))
-        else None
-    }
+  def completion(accumulator: Accumulator = new Accumulator) = {
+    val rc = new ReplCompletion(intp, accumulator)
+    MultiCompletion(shellCompletion, rc)
+  }
+  val shellCompletion = new Completion {
+    override def reset() = ()
+    override def complete(buffer: String, cursor: Int) =
+      if (buffer.startsWith(":")) colonCompletion(buffer, cursor).complete(buffer, cursor)
+      else NoCompletions
+  }
 
   def completionsCommand(what: String): Result = {
-    val completions = newCompleter().complete(what, what.length)
+    val completions = in.completion.complete(what, what.length)
     val prefix = if (completions == NoCompletions) "" else what.substring(0, completions.cursor)
-
-    val completionLines =
-      completions.candidates.map { c =>
-        s"[completions] $prefix$c"
-      }
-
-    if (completionLines.nonEmpty) {
-      echo(completionLines.mkString("\n"))
-    }
-
+    completions.candidates.map(c => s"[completions] $prefix$c").foreach(echo)
```

Comment:
This needs to be reverted.

Echo flushes between lines and there is a huge difference whether you flush after each line or only once.

I remember writing the code first as you are now proposing and changing to what it is currently in master because of a good reason.

Reply:
d1f4a72cf9821a22b6705ad46b4ad47c261c2062 LGTY?